controllers/workflow_controller.go

```diff
@@ -27,15 +31,49 @@ func (r *DSPAReconciler) ReconcileWorkflowController(dsp *dspav1.DataSciencePipe
 
 	log := r.Log.WithValues("namespace", dsp.Namespace).WithValues("dspa_name", dsp.Name)
 
-	if dsp.Spec.WorkflowController == nil || !dsp.Spec.WorkflowController.Deploy {
-		log.Info("Skipping Application of WorkflowController Resources")
-		return nil
-	}
+	// Get the management state for the WorkflowController subcomponent from the config
+	// Expected format example: {"managementState":"Managed"}
+	dspoArgoWorkflowsControllersJSON := config.GetStringConfigWithDefault("DSPO.ArgoWorkflowsControllers", config.DefaultArgoWorkflowsControllers)
 
-	log.Info("Applying WorkflowController Resources")
+	argoWorkflowsControllersConfig := map[string]string{}
+	var argoWorkflowsControllerManagementState string
 
-	err := r.ApplyDir(dsp, params, workflowControllerTemplatesDir)
+	err := json.Unmarshal([]byte(dspoArgoWorkflowsControllersJSON), &argoWorkflowsControllersConfig)
 	if err != nil {
+		log.Info(fmt.Sprintf("Unable to parse Argo Workflows Controller management state, using default value: %s", config.DefaultArgoWorkflowsControllersManagementState))
+		argoWorkflowsControllerManagementState = config.DefaultArgoWorkflowsControllersManagementState
+	} else {
+		argoWorkflowsControllerManagementState = argoWorkflowsControllersConfig["managementState"]
+	}
+
+	// Conditionally deploy the WorkflowController resource depending on the speciified management state
+	// Managed (or blank) - deploy the WorkflowController subcomponent
+	// Removed - skip deploying, and remove if already present, the WorkflowController subcomponent
+	// All other values - Invalid configuration, return an error
+	switch argoWorkflowsControllerManagementState {
+	case "Managed", "":
+
+		if dsp.Spec.WorkflowController == nil || !dsp.Spec.WorkflowController.Deploy {
+			log.Info("Skipping Application of WorkflowController Resources")
+			return nil
+		}
+
+		log.Info("Applying WorkflowController Resources")
+
+		err = r.ApplyDir(dsp, params, workflowControllerTemplatesDir)
+		if err != nil {
+			return err
+		}
+
+	case "Removed":
+		log.Info("Removing WorkflowController Resources (if present)")
+		err = r.DeleteResourceDir(params, workflowControllerTemplatesDir)
+		if err != nil {
+			return err
+		}
+
+	default:
+		err = fmt.Errorf("invalid management state for WorkflowController Resources: %s", argoWorkflowsControllerManagementState)
```

Comment:
Should it fallback to `dsp.Spec.WorkflowController.Deploy` in this case?

Reply:
not sure I agree - this is only reachable if the management state is pretty specifically invalid, but still properly formatted, ie {"managementState": "NotARealManagementState"}.  By erroring here we wouldn't _delete_ existing WCs, but also wont deploy new ones, and would leave the DSPA in a Not Ready state, which may be more in-line with the actual state of the DSPA than if we just defaulted to Managed behavior.  

That said, I admittedly originally had the proposed behavior before updating as it is here, so if that's the path we want to go, I can update/revert it.